src/Microsoft.AspNet.Razor/CodeGenerators/CSharpCodeWriter.cs

```diff
@@ -445,18 +458,26 @@ private void WriteVerbatimStringLiteral(string literal)
         {
             Write("@\"");
 
-            foreach (char c in literal)
+            // We need to find the index of each '"' (double-quote) to escape it.
+            var start = 0;
+            int end;
+            while ((end = literal.IndexOf('\"', start)) > -1)
             {
-                if (c == '\"')
+                if (end > start)
```

Comment:
yeah this isn't really needed, because in the case where `end == start` then `count == 0`